# Assignment 3: LinkedIn Campaign Assessment

### Objective

* An advertising agency has completed a campaign on LinkedIn for a customer in high-value services industry.

* The campaign has utilized two placements: **(1) In-feed & (2) Messages**

* Students are tasked with assessing the campaign’s effectiveness to compare these
channels and prepare recommendations for optimizing the campaign.

#### Dataset: Provided by Professor [Ivan Fedorenko](https://www.csueastbay.edu/directory/profiles/mktg/fedorenkoivan.html) | Digital Media Analytics - California State University, East Bay


### Calculate and apply the following metrics:
* Impressions & CPM for building Awareness
* Clicks, CTR & CPC for Interest / Consideration
* Leads, CR & CPA for Action / Conversion

In [2]:
%cd /content/drive/MyDrive/Assignments/Digital Media Analytics/Assignments

/content/drive/MyDrive/Assignments/Digital Media Analytics/Assignments


In [3]:
# Install needed Libraries
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

In [4]:
#Load the data
Linkedin_df = pd.read_csv('Linkedin Ads assessment.xlsx - Sheet1.csv')
Linkedin_df.head()

Placement Impressions       Cost, $    CPM, $  Clicks    CTR  \
0       LinkedIn Ads    230,242    $ 15,281.6        NaN    1763    NaN   
1  LinkedIn Messages      6,986     $ 4,500.0        NaN    2578    NaN   
2              Total    237,228    $ 19,781.6    $ 83.4     4341  1.83%   

        CPC  Leads   CR %     CPA, $  
0       NaN     52    NaN        NaN  
1       NaN      7    NaN        NaN  
2   $ 4.56      59  1.36%   $ 335.3

##Question 1: Campaign Performance vs. Benchmarks

###Calculate CTR (Click-Through-Rate)

In [5]:
# Formula: CTR = (Clicks / Impressions) * 100

In [6]:
# LinkedIn Ads:
Ads = (1763/230242) * 100

# LinkedIn Messages:
Messages = (2578/6986) * 100
Ads, Messages

(0.7657160726539902, 36.902376180933295)

LinkedIn Messages overperforms by **36.90%**, while LinkedIn Ads is underperforming by **0.77%**.


### Calculate CPC (Cost-Per-Conversion)

In [7]:
# Formula: CPC = (Cost / Clicks)

In [8]:
# LinkedIn Ads:
Ads = (15281.57/1763)

# LinkedIn Messages:
Messages = (4500/2578)
Ads, Messages

(8.66793533749291, 1.7455391776570985)

LinkedIn Messages is more cost efficient by **$1.75**, while LinkedIn Ads has a cost of **\$8.67%**, which is too expensive for any campaign.


### Calculate CR (Conversion Rate)

In [9]:
# Formula: CR = (Leads / Clicks) * 100

In [10]:
# LinkedIn Ads:
Ads = (52/1763) * 100

# LinkedIn Messages:
Messages = (7/2578) * 100
Ads, Messages

(2.949517867271696, 0.27152831652443754)

LinkedIn Ads generates a **2.95%** conversion rate, outperforming LinkedIn Messages’ **0.27%** conversion rate. However, both rates fall short of the industry benchmark of **5%**.



##Question 2: Best Placement in Terms of CR and CPA

### Calculating CPA (Cost-Per-Acquisition)

In [11]:
# Formula: CPA = Cost / Leads

In [12]:
# LinkedIn Ads:
Ads = (15281.57/52) * 100

# LinkedIn Messages:
Messages = (4500/7) * 100
Ads, Messages

(29387.634615384617, 64285.71428571429)

Since **LinkeIn Ads has lower CPA and higher CR**, it is the better-performing placement.

##Question 3: Additional Leads from Budget Reallocation

#### New Leads = (Cost of Worst-performaning channel / CPA of best-performing channel) - Exisiting Leads from Worst-performing channel

In [13]:
(4500 / 293.88) - 7

8.312372396896693

**8.31** additonal Leads from budget allocation

##Summary
* CTR: LinkedIn Messages (**36.90%**) overperforms, while LinkedIn Ads (**0.77%**) underperforms.

* CPC: LinkedIn Messages (**\$1.75**) is more cost-efficient than LinkedIn Ads (**$8.67**).

* CR: LinkedIn Ads (**2.95%**) is better than LinkedIn Messages (**0.27%**) but still below the benchmark (**5%**).

* CPA: LinkedIn Ads (**\$293.88**) is more efficient than LinkedIn Messages (**\$642.86**).

Reallocating funds from LinkedIn Messages to LinkedIn Ads could **result in 8 additional leads.**